File: src/main/java/com/eager/questioncloud/coupon/repository/CouponJpaRepository.java
package com.eager.questioncloud.coupon.repository;

import com.eager.questioncloud.coupon.entity.CouponEntity;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CouponJpaRepository extends JpaRepository<CouponEntity, Long> {
    Optional<CouponEntity> findByCode(String code);
}

File: qc-payment/src/main/kotlin/com/eager/questioncloud/coupon/infrastructure/repository/CouponJpaRepository.kt
package com.eager.questioncloud.coupon.infrastructure.repository

import com.eager.questioncloud.coupon.infrastructure.entity.CouponEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface CouponJpaRepository : JpaRepository<CouponEntity, Long> {
    fun findByCode(code: String): Optional<CouponEntity>
}
